lib/apple_formatter.rb

```diff
@@ -46,10 +46,10 @@ def self.write_strings_file(terms, file, substitutions: nil, print_date: false,
         end
 
         definition = Poesie::process(definition, substitutions)
-                    .gsub("\u2028", '') # Sometimes inserted by the POEditor exporter
-                    .gsub("\n", '\n') # Replace actual CRLF with '\n'
-                    .gsub('"', '\\"') # Escape quotes
-                    .gsub(/%(\d+\$)?s/, '%\1@') # replace %s with %@ for iOS
+                    context.gsub("\u2028", '') # Sometimes inserted by the POEditor exporter
+                    context.gsub("\n", '\n') # Replace actual CRLF with '\n'
+                    context.gsub('"', '\\"') # Escape quotes
+                    context.gsub(/%(\d+\$)?s/, '%\1@') # replace %s with %@ for iOS
```

Comment:
This means that those substitutions are now done on the context, while they were done on the definition before, so this change the feature & behavior, right?

Reply:
I've looked into this a bit more, and it seems like the errors went away because the sub's are not executed anymore... 😳 So this change is not an option since quotes, CRLF, etc, won't be escaped. Any idea's of why Asian chars will break the script?

Reply:
Maybe that's because asian characters are higher in the Unicode plane, so have higher Unicode codepoints and if the text isn't encoded in the proper encoding, `gsub` might either fail or misinterpret the bytes/codepoints during replacement and mess up the string

Maybe try forcing the `definition` to `'utf-8'` before applying the `.gsub` chain could help?

```
definition = Poesie::process(definition, substitutions)
	                    .encode('utf-8', 'utf-8')
	                    .gsub("\u2028", '') # Sometimes inserted by the POEditor exporter
	                    .gsub("\n", '\n') # Replace actual CRLF with '\n'
	                    .gsub('"', '\\"') # Escape quotes
	                    .gsub(/%(\d+\$)?s/, '%\1@') # replace %s with %@ for iOS
```
See https://stackoverflow.com/a/9560882

Reply:
Thats a valid point! Let me double check once more if there's some issues with the translations 🙂

Reply:
Now I’ve finally found the main issue causing this problem, for example Chinese don’t use plurals, so when exporting terms that have plurals in other languages, there will only be one key/value in the “definition” element. POEditor uses the plural rules defined in Unicode Common Locale Data (found here: http://cldr.unicode.org/index/cldr-spec/plural-rules), that specifies that “other” should be used if the language only has a single form. This means that definition is set to a nil-value, since the script currently uses the key “one” for getting the first plural value.

I have made a new commit to address this and have dropped the previous one :-)

Reply:
Nice investigative work 🕵️‍♀️👍

Reply:
Thanks! 😊